contents/swift/transition-to-swift.md

```diff
@@ -0,0 +1,309 @@
+# Swift
+
+Authors: Ch'ng Ming Shin
+
+# Overview
+
+If you are transitioning from another language to Swift, I am here to make your life easier. The common solutions to "How to learn Swift" from Quora and StackOverflow are to
+
+1. Dive into a project
+
+2. Read the official documentation
+https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/index.html
+
+
+Those are great suggestions but diving into a new project with no prior experience to Swift makes it an ardous trial of google searches of how to implement stuff. And reading documentation, while really helpful, might be overkill and boring as there might be many concepts that you already know.
+
+To ease this pain, I am providing a 3rd solution where I try to summarise the more common and important aspects of Swift that are generally different from other languages.
+
+
+
+# Getting Started
+
+Cheatsheets:
+One page cheatsheet: https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf
+
+Multiple pages of cheatsheet (mainly code examples with comments): https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html
+
+Main Swift Concepts (less indepth, wide range of topics, long explanations): https://guides.codepath.com/ios/Swift-Basics
+
+Basics of Apple Swift in playground: https://github.com/Codeido/swift-basics
+
+Good readings on Swift:
+Design of Swift, why change from obj-C
+http://swiftjester.org/papers/whats-different-about-swift.html
+
+Other Resources:
+http://www.whoishostingthis.com/resources/swift/
+
+Guide to Xcode(IDE for Swift):
+http://codewithchris.com/xcode-tutorial/
+
+# Swift Basics
+
+## [The Absolute Basics](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html)
+
+Things to note:
+
+Constant and variable names can contain almost any character, including Unicode characters
+
+There is the idea of mutability of variables. If a variable should not be immutable, use let to declare it as a immutable constant. (let is similiar to final in Java)
+
+`
+let a = 2
+var b = 3
+`
```

Comment:
I feel that ``` should be used instead of `, otherwise they all become one line when I preview it on GitHub.

Alternatively, put the starting \` and ending \` on each line:
\`let a = 2\`
\`var b = 3\`